config/interpolate_funcs.go

```diff
@@ -1657,3 +1663,74 @@ func interpolationFuncAbs() ast.Function {
 		},
 	}
 }
+
+// interpolationFuncRsaDecrypt implements the "rsadecrypt" function that does
+// RSA decryption.
+func interpolationFuncRsaDecrypt() ast.Function {
+	return ast.Function{
+		ArgTypes:   []ast.Type{ast.TypeString, ast.TypeString},
+		ReturnType: ast.TypeString,
+		Callback: func(args []interface{}) (interface{}, error) {
+			s := args[0].(string)
+			key := args[1].(string)
+
+			b, err := base64.StdEncoding.DecodeString(s)
+			if err != nil {
+				b = []byte(s)
+			}
+
+			block, _ := pem.Decode([]byte(key))
+			if block == nil {
+				return "", fmt.Errorf("Failed to read key %q: no key found", key)
+			}
+			if block.Headers["Proc-Type"] == "4,ENCRYPTED" {
+				return "", fmt.Errorf(
+					"Failed to read key %q: password protected keys are\n"+
+						"not supported. Please decrypt the key prior to use.", key)
+			}
+
+			x509Key, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+			if err != nil {
+				return "", err
+			}
+
+			out, err := rsa.DecryptPKCS1v15(nil, x509Key, b)
+			if err != nil {
+				return "", err
+			}
+
+			return string(out), nil
+		},
+	}
+}
+
+// interpolationFuncReplace implements the "rsaencrypt" function that does
+// RSA encryption.
+func interpolationFuncRsaEncrypt() ast.Function {
+	return ast.Function{
+		ArgTypes:   []ast.Type{ast.TypeString, ast.TypeString},
+		ReturnType: ast.TypeString,
+		Callback: func(args []interface{}) (interface{}, error) {
+			s := args[0].(string)
+			key := args[1].(string)
+
+			block, _ := pem.Decode([]byte(key))
+			if block == nil {
+				return "", fmt.Errorf("Failed to read key %q: no key found", key)
+			}
+
+			x509Key, err := x509.ParsePKIXPublicKey(block.Bytes)
+			if err != nil {
+				return "", err
+			}
+
+			random := rand.Reader
+			out, err := rsa.EncryptPKCS1v15(random, x509Key.(*rsa.PublicKey), []byte(s))
```

Comment:
The docstring for `EncryptPKCS1v15` contains this warning:

> WARNING: use of this function to encrypt plaintexts other than session keys is dangerous. Use RSA OAEP in new protocols.

I don't have enough context to know whether that's important here, but since this is a crypto sort of thing I worry that we may end up passing whatever the limitation is on to Terraform users without disclosing it, which may cause problems for users down the line. I'd like to take some time to research better what the situation is before moving foward with this.